src/Windows/R/Components/Impl/InteractiveWorkflow/Implementation/InteractiveWindowWriter.cs

```diff
@@ -26,16 +26,33 @@ internal sealed class InteractiveWindowWriter : IDisposable {
         private volatile bool _disposed;
 
         private int _currentLineStart = -1;
+        private int _lastInteractiveWindowLineCount;
         private string _newText;
 
         public InteractiveWindowWriter(IMainThread mainThread, IInteractiveWindow interactiveWindow) {
             _mainThread = mainThread;
             _interactiveWindow = interactiveWindow;
+            _interactiveWindow.TextView.TextBuffer.Changed += OnTextViewBufferChanged;
```

Comment:
`_interactiveWindow.TextView.TextBuffer` - let's cache this text buffer for dispose. Just in case it is changed or `Dispose` is called from non-UI thread